keras_tuner/tuners/gridsearch.py

```diff
@@ -0,0 +1,265 @@
+    """Grid search oracle.
+
+    Args:
+        objective: A string, `keras_tuner.Objective` instance, or a list of
+            `keras_tuner.Objective`s and strings. If a string, the direction of
+            the optimization (min or max) will be inferred. If a list of
+            `keras_tuner.Objective`, we will minimize the sum of all the
+            objectives to minimize subtracting the sum of all the objectives to
+            maximize. The `objective` argument is optional when
+            `Tuner.run_trial()` or `HyperModel.fit()` returns a single float as
+            the objective to minimize.
+        max_trials: Optional integer, the total number of trials (model
+            configurations) to test at most. Note that the oracle may interrupt
+            the search before `max_trial` models have been tested if the search
+            space has been exhausted. If left unspecified, it runs till the
+            search space is exhausted.
+        seed: Optional integer, the random seed.
+        hyperparameters: Optional `HyperParameters` instance. Can be used to
+            override (or register in advance) hyperparameters in the search
+            space.
+        tune_new_entries: Boolean, whether hyperparameter entries that are
+            requested by the hypermodel but that were not specified in
+            `hyperparameters` should be added to the search space, or not. If
+            not, then the default value for these parameters will be used.
+            Defaults to True.
+        allow_new_entries: Boolean, whether the hypermodel is allowed to
+            request hyperparameter entries not listed in `hyperparameters`.
+            Defaults to True.
+    """
+
+    def __init__(
+        self,
+        objective=None,
+        max_trials=None,
+        seed=None,
+        hyperparameters=None,
+        allow_new_entries=True,
+        tune_new_entries=True,
+    ):
+        super(GridSearchOracle, self).__init__(
+            objective=objective,
+            max_trials=max_trials,
+            hyperparameters=hyperparameters,
+            tune_new_entries=tune_new_entries,
+            allow_new_entries=allow_new_entries,
+            seed=seed,
+        )
+
+    def populate_space(self, trial_id):
+        """Fill the hyperparameter space with values.
+
+        Args:
+            trial_id: A string, the ID for this Trial.
+
+        Returns:
+            A dictionary with keys "values" and "status", where "values" is
+            a mapping of parameter names to suggested values, and "status"
+            is the TrialStatus that should be returned for this trial (one
+            of "RUNNING", "IDLE", or "STOPPED").
+        """
+        values = self._exhaustive_values()
+        if values is None:
+            return {"status": trial_module.TrialStatus.STOPPED, "values": None}
+        return {"status": trial_module.TrialStatus.RUNNING, "values": values}
+
+    def create_trial(self, tuner_id):
```

Comment:
We should move the logics you added in `create_trial` into `populate_space` to do them on the fly.
Prefer to keep `create_trial` untouched unless we really need to.